### PR TITLE
Flake fix: patch probe CR with an incrementing annotation

### LINKED ISSUE #78913: TestWebhookConverter storage version wait is flaky
**Which jobs are failing**:

pull-kubernetes-integration

https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&test=TestWebhookConverter

**Which test(s) are failing**:

TestWebhookConverterWithDefaulting

/assign @sttts
/priority important-soon
/area custom-resources
/sig api-machinery
/kind flake
/milestone v1.16

### PR DESCRIPTION
**Pull request by roycaihw**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
Mutate the probe CR so it persists to etcd and updates the version in storage. Ref https://github.com/kubernetes/kubernetes/issues/78913#issuecomment-502321159

**Which issue(s) this PR fixes**:

Fixes #78913

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/area custom-resources
/sig api-machinery
/cc @sttts @jpbetz @liggitt

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go [hunk 1]
```diff
@@ -954,10 +954,12 @@ func (c *conversionTestContext) setStorageVersion(t *testing.T, version string)
 }
 
 func (c *conversionTestContext) waitForStorageVersion(t *testing.T, version string, versionedClient dynamic.ResourceInterface, obj *unstructured.Unstructured) *unstructured.Unstructured {
-	if err := c.etcdObjectReader.WaitForStorageVersion(version, obj.GetNamespace(), obj.GetName(), 30*time.Second, func() {
-		if _, err := versionedClient.Patch(obj.GetName(), types.MergePatchType, []byte(`{}`), metav1.PatchOptions{}); err != nil {
+	if err := c.etcdObjectReader.WaitForStorageVersion(version, obj.GetNamespace(), obj.GetName(), 30*time.Second, func(idx int) {
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go]: Nit: can avoid changing the signature by making the count a closure var

**Comment by liggitt** [comment]
Hmm… actually, the no-op write compares the bytes that would be persisted to etcd, which would actually be different if the storage version had changed. I'm not sure this is the actual issue.

**Comment by jpbetz** [comment]
I’ll try to wrap the test in some loops tonight and see if i can reproduce
the flake more consistently.

**Comment by roycaihw** [comment]
/close

I think https://github.com/kubernetes/kubernetes/pull/79114 is likely the cause

**Pull request closed without merging**